compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -2800,8 +2840,78 @@ string t_py_generator::type_name(t_type* ttype) {
   return ttype->get_name();
 }
 
+string t_py_generator::arg_hint(t_type* type) {
+  if (gen_type_hints_) {
+    return ": " + type_to_py_type(type);
+  }
+
+  return "";
+}
+
+string t_py_generator::member_hint(t_type* type, t_field::e_req req) {
+  if (gen_type_hints_) {
+    if (req != t_field::T_REQUIRED) {
+      return ": typing.Optional[" + type_to_py_type(type) + "]";
```

Comment:
i wonder if we can just generate `| None` if python version allows it

Reply:
The union operator is not compatible with python 3.7-3.11 afaik. You can get around that by importing annotations from the future module, but not much of a change at that point.

Reply:
I believe this would work, but only as of 3.10 (which is EOL in a couple months.) Using `typing.*` and `Union` syntax allows us to target all of 3.5 to 3.12+

Reply:
Theres really no benefit to `| None` besides style. I love that syntax too but this is quite convenient for making it work "always" and without any need to add branches.